assets/src/scripts/index.js

```diff
@@ -1,29 +1,21 @@
 // Initialize the 18F Web design standards
 require('uswds');
-const { timeFormat } = require('d3-time-format');
 
-const { getTimeseries } = require('./models');
-const Hydrograph = require('./hydrograph');
 
-// Create a time formatting function from D3's timeFormat
-const formatTime = timeFormat('%c %Z');
+const COMPONENTS = {
+    hydrograph: require('./components/hydrograph').attachToNode,
+    map: require('./components/map').attachToNode
+};
+
 
 function main() {
-    let nodes = document.getElementsByClassName('hydrograph');
+    let nodes = document.getElementsByClassName('wdfn-component');
     for (let node of nodes) {
-        getTimeseries({sites: [node.dataset.siteno]}, series => {
-            let dataIsValid = series[0] && !series[0].values.some(d => d.value === -999999);
-            new Hydrograph({
-                element: node,
-                data: dataIsValid ? series[0].values : [],
-                yLabel: dataIsValid ? series[0].variableDescription : 'No data',
-                title: dataIsValid ? series[0].variableName : '',
-                desc: dataIsValid ? series[0].variableDescription + ' from ' + formatTime(series[0].seriesStartDate) + ' to ' + formatTime(series[0].seriesEndDate) : ''
-            });
-        });
+        COMPONENTS[node.dataset.component](node, node.dataset);
```

Comment:
Components attached to the DOM node here - data attributes passed as component options.